### PR TITLE
Add playout-delay section to the rtp header extensions.

### PR DESCRIPTION
**Pull request by stefanholmer**

### CODE REVIEW

File: experiments/rtp-hdrext/playout-delay/index.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+permalink: /experiments/rtp-hdrext/playout-delay/
+---
+
+Name: "Playout Delay" ; "RTP Header Extension to control Playout Delay"
+
+Formal name: <http://www.webrtc.org/experiments/rtp-hdrext/playout-delay>
+
+SDP "a= name": "playout-delay" ; this is also used in client/cloud signaling.
+
+Introduction
```
Comment (alvestrand) [inline experiments/rtp-hdrext/playout-delay/index.md:13]: Use markdown markup to indicate sections - I think it's "#" and "##" for first and second level headers.

File: experiments/rtp-hdrext/playout-delay/index.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+On WebRTC, the RTP receiver continuously measures inter-packet delay and evaluates packet jitter. Besides this, an estimated delay for decode and render at the receiver is computed. The jitter buffer, the local time extrapolation and the predicted render time (based on predicted decode and render time) impact the delay on a frame before it is rendered at the receiver.
+
+This document proposes an RTP extension to enable the RTP sender to try and limit the amount of playout delay at the receiver in a certain range. A minimum and maximum delay from the sender provides guidance on the range over which the receiver can smooth out rendering.
```
Comment (alvestrand) [inline experiments/rtp-hdrext/playout-delay/index.md:17]: If this is a quote from a document, please insert "Copied from <draftname>" at the top.

File: experiments/rtp-hdrext/playout-delay/index.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+SDP "a= name": "playout-delay" ; this is also used in client/cloud signaling.
+
+Introduction
+
+On WebRTC, the RTP receiver continuously measures inter-packet delay and evaluates packet jitter. Besides this, an estimated delay for decode and render at the receiver is computed. The jitter buffer, the local time extrapolation and the predicted render time (based on predicted decode and render time) impact the delay on a frame before it is rendered at the receiver.
+
+This document proposes an RTP extension to enable the RTP sender to try and limit the amount of playout delay at the receiver in a certain range. A minimum and maximum delay from the sender provides guidance on the range over which the receiver can smooth out rendering. 
+
+Thus, this extension, aims to provide sender’s intent to the receiver on how quickly a frame needs to be rendered.
```
Comment (alvestrand) [inline experiments/rtp-hdrext/playout-delay/index.md:19]: Nit: superfluous comma after "extension".

File: experiments/rtp-hdrext/playout-delay/index.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+0: Certain gaming scenarios (likely without audio) where we will want to play the frame as soon as possible. Also, for remote desktop without audio where rendering a frame asap makes sense
+
+100/150/200: These could be the max target latency for interactive streaming use cases depending on the actual application (gaming, remoting with audio, interactive scenarios)
+
+400: Application that want to ensure a network glitch has very little chance of causing a freeze can start with a minimum delay target that is high enough to deal with network issues. Video streaming is one example.
+
+The header is attached on the RTP packet by the RTP sender when it needs to change the min and max smoothing delay at the receiver. Once the sender confirms the receipt of at least one RTP packet which has the min and max details, it MAY stop providing details on all further RTP packets until another change warrants communicating the details to the receiver again.  This is done as follows:
```
Comment (alvestrand) [inline experiments/rtp-hdrext/playout-delay/index.md:54]: Nit: I think it is the receiver that confirms the receipt of the packet, not the sender. Alternatively, "Once the sender is informed that at least one RTP packet which has the min and max details is delivered".

File: experiments/rtp-hdrext/playout-delay/index.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+RTCP feedback to RTP sender includes the highest sequence number that was seen on the RTP receiver. The RTP sender can track the sequence number on the packet that first had the playout delay extension and then stop sending the extension once the received sequence number is greater than the sequence number on the first packet containing playout delay.
```
Comment (alvestrand) [inline experiments/rtp-hdrext/playout-delay/index.md:56]: "containing playout delay" -> "containing the current values for playout delay in this extension".

**Review by alvestrand — COMMENTED** [review]
Looks good, but I always find nits :-)

**Comment by KaptenJansson** [comment]
@stefanholmer did you use two different emails or not sign the CLA with all your emails?

**Comment by KaptenJansson** [comment]
Ping pong

**Comment by KaptenJansson** [comment]
pong ping

**Comment by stefanholmer** [comment]
I signed it!

**Comment by stefanholmer** [comment]
I think...

**Comment by KaptenJansson** [comment]
Do you plan on addressing the nits?

FYI the error on travis is due to the experiment links to this page which has not been published yet hence why it fails (it checks that all links are working)

**Comment by alvestrand** [comment]
@stefanholmer I found this lying around - do you plan to merge it?

**Comment by benbro** [comment]
Can playout-delay work on audio too?
Can we lip-sync delayed video and audio of the same stream?

**Comment by KaptenJansson** [comment]
Ping?

**Comment by drkron** [comment]
I have been assigned to take over this PR. However, I didn't manage to add changes here so instead I've create a new PR where the review comments above have been addressed. This means that this PR can be abandoned.

See https://github.com/webrtc/webrtc-org/pull/216 for the new PR.

**Comment by alvestrand** [comment]
Closing PR as per last comment.

**Pull request closed without merging**